File: src/main/java/com/procurement/orchestrator/domain/enums/AwardCriteria.java
package com.procurement.orchestrator.domain.enums;

import com.procurement.orchestrator.exception.EnumException;

import java.util.Arrays;
import java.util.Map;
import java.util.stream.Collectors;

public enum AwardCriteria {
    COST_ONLY("costOnly"),
    QUALITY_ONLY("qualityOnly"),
    RATED_CRITERIA("ratedCriteria"),
    PRICE_ONLY("priceOnly");

    private String value;

    private static Map<String, AwardCriteria> elements = Arrays.stream(values())
            .collect(Collectors.toMap(
                    item -> item.value.toUpperCase(),
                    item -> item)
            );

    AwardCriteria(String value) {
        this.value = value;
    }

    @Override
    public String toString() {
        return this.value;
    }

    public static AwardCriteria fromString(String value) {
        final AwardCriteria element = elements.get(value.toUpperCase());
        if (element != null) {
            return element;
        } else {
            throw new EnumException(AwardCriteria.class.getCanonicalName(), value, Arrays.toString(values()));
        }

    }
}

File: src/main/kotlin/com/procurement/orchestrator/domain/model/tender/AwardCriteria.kt
package com.procurement.orchestrator.domain.model.tender

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonValue
import com.procurement.orchestrator.domain.EnumElementProvider

enum class AwardCriteria(@JsonValue override val key: String) : EnumElementProvider.Key {

    COST_ONLY("costOnly"),
    PRICE_ONLY("priceOnly"),
    QUALITY_ONLY("qualityOnly"),
    RATED_CRITERIA("ratedCriteria");

    override fun toString(): String = key

    companion object : EnumElementProvider<AwardCriteria>(info = info()) {

        @JvmStatic
        @JsonCreator
        fun creator(name: String) = orThrow(name)
    }
}
